### PR TITLE
Cancer screening history survey

### PR DESCRIPTION
**Pull request by we-ai**
Updates in this PR:
- Implement requirements of cancer screening history survey ([issue #1145](https://github.com/episphere/connect/issues/1145))
- Fix delayed app version refreshing ([issue #1171](https://github.com/episphere/connect/issues/1171))

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,22 +25,30 @@ let saveFlag = false;
...
-        return;
-    });
-
-    navigator.serviceWorker.ready.then(() => {
-        if (navigator.serviceWorker.controller) {
-            navigator.serviceWorker.controller.postMessage({ action: "getAppVersion" });
+  navigator.serviceWorker
+    .register("./serviceWorker.js")
+    .then((registration) => {
+      registration.onupdatefound = () => {
```
Comment (JoeArmani) [inline index.js:31]: Nice. I just read up on this.

**Review by JoeArmani — APPROVED** [review]
looks good

File: index.js [hunk 2]
```diff
@@ -25,22 +25,30 @@ let saveFlag = false;
...
+        const sw = registration.installing;
+        if (sw) {
+          sw.onstatechange = () => sw.state === "activated" && sw.postMessage({ action: "getAppVersion" });
         }
+      };
+    })
+    .catch((err) => {
+      console.error("Service worker registration failed.", err);
```
Comment (Gbarra9) [inline index.js:39]: I like the error handling you added

**Review by Gbarra9 — APPROVED** [review]
Looks great!

**Pull request merged**